Sources_v3/Controllers/MessageControllerGeneric.swift

```diff
@@ -0,0 +1,201 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+public extension Client {
+    func messageController(messageId: MessageId) -> MessageControllerGeneric<ExtraData> {
+        .init(client: self, messageId: messageId, environment: .init())
```

Comment:
nit: Environment should always have a default value. We change it only in tests, ideally we don't touch it in production code at all.